ballerina/client.bal

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+# Consists of APIs to integrate with LDAP.
+public isolated client class Client {
+
+    # Gets invoked to initialize the LDAP client.
+    #
+    # + config - The configurations to be used when initializing the client
+    # + return - An error if client initialization failed
+    public isolated function init(*ConnectionConfig config) returns Error? {
+        check self.initLdapConnection(config);
+    }
+
+    private isolated function initLdapConnection(ConnectionConfig config) returns Error? = @java:Method {
+        'class: "io.ballerina.lib.ldap.Ldap"
+    } external;
+
+    # Creates an entry in a directory server.
+    #
+    # + distinguishedName - The distinguished name of the entry
+    # + entry - The information to add
+    # + return - `error` if the operation fails or `()` if successfully updated
+    remote isolated function add(string distinguishedName, record {|anydata...;|} entry)
+        returns LDAPResponse|Error = @java:Method {
+        'class: "io.ballerina.lib.ldap.Ldap"
+    } external;
+
+    # Removes an entry in a directory server.
+    #
+    # + distinguishedName - The distinguished name of the entry to remove
+    # + return - `error` if the operation fails or `()` if successfully updated
+    remote isolated function delete(string distinguishedName) returns LDAPResponse|Error = @java:Method {
```

Comment:
Shouldn't this `LDAPResponse` be `LdapResponse`?

Reply:
Fixed in [37aaf59](https://github.com/ballerina-platform/module-ballerina-ldap/pull/45/commits/37aaf59ce89d002eb5bd0b08a8586dbbb2bd0665)

Reply:
Fixed in [cdde708](https://github.com/ballerina-platform/module-ballerina-ldap/pull/45/commits/cdde70852f32464ef384105c5366885fa9486d98)